github/util_v4_branch_protection.go

```diff
@@ -434,6 +499,16 @@ func setPushes(protection BranchProtectionRule, data BranchProtectionResourceDat
 	return pushActors
 }
 
+func setForcePushBypassers(protection BranchProtectionRule, data BranchProtectionResourceData, meta interface{}) []string {
+	if protection.AllowsForcePushes {
```

Comment:
Unclear how `AllowsForcePushes` and force push bypassers work. Documentation is lacking.

Does `AllowsForcePushes` need to be true for the force push bypassers to work, or are they mutually exclusive?

<img width="513" alt="Screenshot 2023-02-01 at 5 34 42 PM" src="https://user-images.githubusercontent.com/352251/216209589-bb1662e3-be3d-468c-8532-1749fd718a20.png">

Reply:
There are other properties we've seen issues like this with; I'm thinking of branch protection rules settings where it's possible to get into an inconsistent state in which branches are required to be up to date before merging but status checks aren't required to pass. The UI makes such changes impossible but using the API can get you into that goofy state. 

Picture of a consistent state below: 

![image](https://user-images.githubusercontent.com/9327688/225407575-7343898b-ae88-4aa6-a2bd-05a605689219.png)

In the branch protection case, when the "secondary" checkbox is checked but not the "primary", the API behaves as if neither checkbox is checked. It's possible that this case would behave the same way. 

Regardless, I think we should treat this behavior as it should be (IMO) where "Allow force pushes" _does_ need to be true in order for force push bypassers to work.

Reply:
With that, is the assumption that if "Allow force pushes" is true and the specific list of force pushers is empty, then everybody can force push?

Reply:
That is a fantastic question. I would imagine/hope that the API would disallow that state, but I'm not sure how it actually works.